src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
+
+
+    private T[] array = (T[]) new Object[def_capacity];
+    private int size;
+    private int cursor;
+
+
+    private String errorMsg(int index) {
+        return "Index: " + index + ", Size: " + size;
+    }
```

Comment:
There are two notions on how to structure methods within the class: 
1) Put all private/protected methods at the end of the class and leave the public methods on top 
2) Try to keep methods in groups next to places where they are used 

In this case, the first thing I see is a private method which does not really carry that much importance to be displayed first, I would move it to the bottom